### PR TITLE
BL-856-approved-gh-issue-mentor-application-payload-needs-to-send-the-correct-object

### PR DESCRIPTION
**Pull request by nathancai97**
## Description

--fixed "other_info", "job_help", "industry_knowledge", "pair_programming" , "tech_stack" and "commitment" keys to get the correct payload for the mentor application. 

Fixes # (856)(https://bloomtechlabs.atlassian.net/browse/BL-856)

## Loom Video

(https://www.loom.com/share/3a8688a9428d4c40aecd16f39bc9fe7e)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have removed unnecessary comments/console logs from my code
- [x] I have made corresponding changes to the documentation if necessary (optional)
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] No duplicate code left within changed files
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

**Review by MelodyLayne — APPROVED** [review]

**Review by miguelaledesma — APPROVED** [review]
Nice Job on lining up the mentor field with the payload. Really like the refactor on the radio button. Good Job Nathan, code is clean, no changes requested from me.

File: src/components/pages/RoleSignup/Applications/Mentor.js [hunk 1]
```diff
@@ -415,18 +419,14 @@ const Mentor = ({ dispatch, error, successPage }) => {
...
-                    >
-                      Yes
-                    </Radio>
-                    <Radio
-                      style={{ margin: '.2rem', width: '100%' }}
-                      value={'no'}
-                    >
-                      No
-                    </Radio>
+                    {commitmentArray.map(object => (
```
Comment (albert-szabo) [inline src/components/pages/RoleSignup/Applications/Mentor.js:422]: Good usage of a programmatic solution to improve on an earlier verbose, hard-coded version!

**Review by albert-szabo — APPROVED** [review]
This is a solid incremental improvement to the code base. Well done!

**Review by stevenhastings — APPROVED** [review]
I commend you for your determination on what seemed to be a pretty "squirrely" problem.  It's not obvious to me what would have been the cause but, you dug in and rooted out the solution.  Great job! I also appreciate the pointing out of a future problem at the end of your video to let others in on concurrent solutions to, perhaps, a larger issue.  As far as my approval goes, you certainly have it.

File: src/components/pages/RoleSignup/Applications/Mentor.js [hunk 2]
```diff
@@ -365,8 +369,8 @@ const Mentor = ({ dispatch, error, successPage }) => {
                   >
                     {contributionArray.map(area => (
                       <Checkbox
-                        value={area.value}
-                        onValueChange={handleTechStack}
+                        value={area.name}
+                        onChange={e => handleChange(e, 'checkbox')}
```
Comment (ElijahHopkin) [inline src/components/pages/RoleSignup/Applications/Mentor.js:373]: I don't fully understand how this worked before--without the hooks. Thanks for catching this.

File: src/components/pages/RoleSignup/Applications/Mentor.js [hunk 3]
```diff
@@ -474,14 +474,13 @@ const Mentor = ({ dispatch, error, successPage }) => {
                     icon: <InfoCircleOutlined />,
                   }}
                 ></Form.Item>
-                <Form.Item
+                <Input.TextArea
```
Comment (ElijahHopkin) [inline src/components/pages/RoleSignup/Applications/Mentor.js:477]: good catch. This SHOULD mirror the mentee issue so thanks for putting them in alignment

**Review by ElijahHopkin — APPROVED** [review]
can we clean up the console.log on line 62 as part of this push?

**Pull request merged**